modules/react/common/stories/Responsive.stories.mdx

```diff
@@ -0,0 +1,314 @@
+import {ResponsiveContainer} from './examples/ResponsiveStyles';
+import {ResponsiveViewport} from './examples/ResponsiveViewport';
+
+<Meta title="Hooks and Utilities/Responsive Styling" />
+
+# Responsive Styling
+
+Responsive styling is an approach to web development or a set of best practices, used to create a
+layout that can respond to the device's viewport size.
+
+You can read more about the history of
+[Responsive Design](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Responsive_Design).
+
+## Container and Viewport Based Responsive Styling
+
+We will take a look at Container-based and Viewport-based Responsive Styling, what they are and
+which to use.
+
+### Container Based Responsive Styling
+
+With viewport-based responsive styling (media queries), the user's viewport (user's visible area of
+a web page) is measured and styles are applied based on the size of that viewport. We will do the
+same with Container Based Responsive Styling but, we will just be measuring the size of the
+container the element is in instead of the user's viewport.
+
+In the example, we see that there is a `maxWidth` set on the Box component of `1000`. This will set
+the maximum width of that element at `1000px`. What we can do with Container Based Responsive
+Styling is measure when that container's width reaches certain breakpoints
+`(i.e. 1440px, 1024px, 768px, 320px, 0)` and apply styles based on those breakpoints. If the
+container's width were to reach `768px`, we can apply styles to that breakpoint.
+
+```tsx
+<Box maxWidth={1000}>Hello World</Box>
+```
+
+In the interactive example below, when the dropdown's value is updated, the container size will
+change to match the size shown in the dropdown. This is just an example of how container-based
+styles work and will be explained more in depth below.
+
+<ExampleCodeBlock code={ResponsiveContainer} />
+
+### Applying Container-Based Responsive Styling
+
+For container-based responsive styling, we have utilities that will allow you to apply responsive
+styles at those breakpoints. There are four utilities/hooks:
+`useResizeObserver | useResponsiveContainerStyles`
+
+`useResizeObserver` - This is a hook that will provide the width of the container. This can be used
+to measure container size as well.
+
+`useResponsiveContainerStyles` - This is a hook that will allow you to create container-based
+responsive styles with objects (as you can see in the example below). This hook accepts three
+arguments with the third being optional (Style Objects, Container Width, Theme). In each style
+object, there are five sizes that it will accept: `zero, small, medium, large and extra large`.
+These give sizes represent the breakpoints that are at the bottom of this page. The sizes will act
+like `min-width`. For example, if you want to apply styles from `medium` and up, then you would
+write those styles under `m`.
```

Comment:
I don't think we need the `### Applying Container-Based Responsive Styling` here. Let's just get right to explaining the hooks.

```suggestion
This example uses the following hooks:

- `useResizeObserver` provides the width (and height) of the container.

- `useResponsiveContainerStyles` allows you to create container-based responsive styles with
  objects. This hook accepts three arguments: style objects, container width, and theme (optional).
  Each style object accepts keys corresponding to the following sizes: `zero`, `s`, `m`, `l` and
  `xl`. These sizes represent [Canvas breakpoints](#canvas-breakpoints). The sizes act like
  `min-width`. For example, if you want to apply styles from `medium` and up, then you would write
  those styles under `m`.
```

Reply:
Agreed